src/main/java/nl/knaw/dans/migration/core/DataverseLoader.java

```diff
@@ -102,8 +103,8 @@ public void loadFromDataset(String doi, Mode mode) {
                     return;
                 if (mode.doDatasets()) {
                     ActualDataset actualDataset = new ActualDataset();
-                    actualDataset.setMajorVersionNr(v.getVersionNumber());
-                    actualDataset.setMinorVersionNr(v.getVersionMinorNumber());
+                    OptionalInt.of(v.getVersionNumber()).ifPresent(actualDataset::setMajorVersionNr);
+                    OptionalInt.of(v.getVersionMinorNumber()).ifPresent(actualDataset::setMinorVersionNr);
```

Comment:
It would be safer and a more general solution to use objects rather than primitives for https://github.com/DANS-KNAW/dd-verify-migration/blob/a726352d0e69c2901f88d3c1142dbaec1fef8a9c/src/main/java/nl/knaw/dans/migration/core/tables/ActualDataset.java#L46-L56